Update haproxy doc for prometheus endpoint (#8081)
DataDog/integrations-core: haproxy/README.md

#### Prepare HAProxy

##### Using Prometheus

> The check supports a newer implementation starting with HAProxy version 2 (enterprise version 1.9rc1) that is based on a Prometheus endpoint:

Reviewer: ```suggestion
The recommended way to setup this integration is by enabling the Prometheus endpoint on HAProxy. This endpoint is built into HAProxy starting version 2 (enterprise version 1.9rc1). If you are using an older version, consider setting up the [HAProxy Prometheus exporter][17], or alternatively setup the legacy socket-based integration described in the next section.
```

Author: I agree this is easier to understand 👍

1. Configure your `haproxy.conf` using the [official guide][16].

2. [Enable](#configuration) the setting `use_prometheus` in `haproxy.d/conf.yaml`.

> 3. [Restart HAProxy to enable the stats endpoint][3].

Reviewer: ```suggestion
3. [Restart HAProxy to enable the Prometheus endpoint][3].
```

Also, is there a "Restart the Agent" step missing, like we usually have?

> Note: The prometheus endpoint can also be set in prior version using the [HAProxy exporter][17]

Reviewer: This is a bit confusing. The first sentence in this section says "this applies 2.0+ or Enterprise 1.9.1rc1+", but here we say "… but if you're using an older version, this can also work if you install the exporter yourself".

##### Using the stats endpoint

> The Agent collects metrics via a stats endpoint:

Reviewer: I think we should add a comment that this is not the recommended way to setup the integration anymore, but rather still present as a legacy option for existing users. Something like this:

```suggestion
Note: this configuration strategy is provided as a reference for legacy users. If you are setting up the integration for the first time, consider using the Prometheus-based strategy described in the previous section.

The Agent collects metrics via a stats endpoint:
```

#### Prepare HAProxy

##### Using Prometheus

> The recommended way to setup this integration is by enabling the Prometheus endpoint on HAProxy. This endpoint is built into HAProxy starting version 2 (enterprise version 1.9rc1). If you are using an older version, consider setting up the [HAProxy Prometheus exporter][17], or alternatively setup the legacy socket-based integration described in the next section.

Reviewer 2: ```suggestion
The recommended way to set up this integration is by enabling the Prometheus endpoint on HAProxy. This endpoint is built into HAProxy starting with version 2 (enterprise version 1.9rc1). If you are using an older version, consider setting up the [HAProxy Prometheus exporter][17], or alternatively set up the legacy socket-based integration described in the next section.
```

1. Configure your `haproxy.conf` using the [official guide][16].

2. [Enable](#configuration) the setting `use_prometheus` in `haproxy.d/conf.yaml`.

3. [Restart HAProxy to enable the Prometheus endpoint][3].

4. [Restart the Agent][6].

##### Using the stats endpoint

> Note: this configuration strategy is provided as a reference for legacy users. If you are setting up the integration for the first time, consider using the Prometheus-based strategy described in the previous section.

Reviewer 2: ```suggestion
**Note**: This configuration strategy is provided as a reference for legacy users. If you are setting up the integration for the first time, consider using the Prometheus-based strategy described in the previous section.
```
